### PR TITLE
Record view stacks if running anomaly mode

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103185

Now, when you do an inplace mutation and the view is naughty, you get this message:

```
RuntimeError: A view was created in no_grad mode and is being modified inplace with grad mode enabled. Given that this use case is ambiguous and error-prone, it is forbidden. You can clarify your code by moving both the view and the inplace either both inside the no_grad block (if you don't want the inplace to be tracked) or both outside (if you want the inplace to be tracked). To find out where this view was allocated, run your entire forward region under anomaly mode (torch.autograd.detect_anomaly(check_nan=False)).
```

When you run under anomaly mode, you get:

```
RuntimeError: A view was created in no_grad mode and is being modified inplace with grad mode enabled. Given that this use case is ambiguous and error-prone, it is forbidden. You can clarify your code by moving both the view and the inplace either both inside the no_grad block (if you don't want the inplace to be tracked) or both outside (if you want the inplace to be tracked). This view was allocated at:
  File "/data/users/ezyang/c/pytorch/test/test_autograd.py", line 4299, in arglebargle
  File "/data/users/ezyang/c/pytorch/test/test_autograd.py", line 4306, in test_anomaly_gives_view_stack
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/case.py", line 549, in _callTestMethod
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/case.py", line 591, in run
  File "/data/users/ezyang/c/pytorch/torch/testing/_internal/common_utils.py", line 2266, in _run_with_retry
  File "/data/users/ezyang/c/pytorch/torch/testing/_internal/common_utils.py", line 2337, in run
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/case.py", line 650, in __call__
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/suite.py", line 122, in run
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/suite.py", line 84, in __call__
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/suite.py", line 122, in run
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/suite.py", line 84, in __call__
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/runner.py", line 184, in run
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/main.py", line 271, in runTests
  File "/home/ezyang/local/c/pytorch-env/lib/python3.10/unittest/main.py", line 101, in __init__
  File "/data/users/ezyang/c/pytorch/torch/testing/_internal/common_utils.py", line 894, in run_tests
  File "/data/users/ezyang/c/pytorch/test/test_autograd.py", line 11209, in <module>
```

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by zdevito — APPROVED** [review]
Looks good. I think it's fine to do the string formatting there.

File: torch/csrc/autograd/variable.cpp [hunk 1]
```diff
@@ -59,6 +66,14 @@ DifferentiableViewMeta::DifferentiableViewMeta(
   }
 }
 
+void DifferentiableViewMeta::set_creation_meta(CreationMeta new_creation_meta) {
+  TORCH_CHECK(
+      has_bw_view(), "creation_meta can only exist for backward views.");
+  creation_meta_ = new_creation_meta;
+  creation_traceback_ = torch::CapturedTraceback::gather(
```
Comment (soulitzer) [inline torch/csrc/autograd/variable.cpp:73]: Should we check for anomaly mode in this case as well?

Comment (ezyang) [reply torch/csrc/autograd/variable.cpp:73]: yes

**Comment by izaitsevfb** [comment]
@pytorchbot revert -m "Breaks internal builds, see D46629734" -c ghfirst

```
stderr: ld.lld: error: undefined symbol: torch::jit::currentCallstack()
>>> referenced by combined_traceback.cpp:23 

clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Pull request closed without merging**